test/src/main/java/org/apache/accumulo/test/functional/IteratorMincClassCastBugIT.java

```diff
@@ -0,0 +1,516 @@
+        if (vb1[i] != vb2[i]) {
+          return String.format("first diff at offset %,d 0x%02x != 0x%02x", i, 0xff & vb1[i],
+              0xff & vb2[i]);
+        }
+      }
+
+      return null;
+    }
+
+    @Override
+    public Key getTopKey() {
+      return topKey;
+    }
+
+    @Override
+    public Value getTopValue() {
+      return topValue;
+    }
+
+    @Override
+    public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment env) {
+      ChunkCombiner cc = new ChunkCombiner();
+      try {
+        cc.init(source.deepCopy(env), null, env);
+      } catch (IOException e) {
+        throw new IllegalArgumentException(e);
+      }
+      return cc;
+    }
+  }
+
+  /**
+   * Copied from 2.0 Examples. A utility for merging visibilities into the form
+   * {@code (VIS1)|(VIS2)|...|(VISN)}. Used by the ChunkCombiner.
+   */
+  public static class VisibilityCombiner {
+
+    private TreeSet<String> visibilities = new TreeSet<>();
+
+    void add(ByteSequence cv) {
+      if (cv.length() == 0)
+        return;
+
+      int depth = 0;
+      int offset = 0;
+
+      for (int i = 0; i < cv.length(); i++) {
+        switch (cv.byteAt(i)) {
+          case '(':
+            depth++;
+            break;
+          case ')':
+            depth--;
+            if (depth < 0)
+              throw new IllegalArgumentException("Invalid vis " + cv);
+            break;
+          case '|':
+            if (depth == 0) {
+              insert(cv.subSequence(offset, i));
+              offset = i + 1;
+            }
+
+            break;
+        }
+      }
+
+      insert(cv.subSequence(offset, cv.length()));
+
+      if (depth != 0)
+        throw new IllegalArgumentException("Invalid vis " + cv);
+
+    }
+
+    private void insert(ByteSequence cv) {
+      for (int i = 0; i < cv.length(); i++) {
+
+      }
+
+      String cvs = cv.toString();
+
+      if (cvs.charAt(0) != '(')
+        cvs = "(" + cvs + ")";
+      else {
+        int depth = 0;
+        int depthZeroCloses = 0;
+        for (int i = 0; i < cv.length(); i++) {
+          switch (cv.byteAt(i)) {
+            case '(':
+              depth++;
+              break;
+            case ')':
+              depth--;
+              if (depth == 0)
+                depthZeroCloses++;
+              break;
+          }
+        }
+
+        if (depthZeroCloses > 1)
+          cvs = "(" + cvs + ")";
+      }
+
+      visibilities.add(cvs);
+    }
+
+    byte[] get() {
+      StringBuilder sb = new StringBuilder();
+      String sep = "";
+      for (String cvs : visibilities) {
+        sb.append(sep);
+        sep = "|";
+        sb.append(cvs);
+      }
+
+      return sb.toString().getBytes();
+    }
+  }
+
+  public static final byte[] nullbyte = new byte[] {0};
+
+  /**
+   * Join some number of strings using a null byte separator into a text object.
+   *
+   * @param s
+   *          strings
+   * @return a text object containing the strings separated by null bytes
+   */
```

Comment:
This could probably just be a one-line inline comment. We're not publishing javadocs for ITs. This method can probably also be private... as could some others added in this file for testing.